lib/spack/llnl/util/filesystem.py

```diff
@@ -1855,15 +1910,7 @@ def prefixes(path):
     """
     if not path:
         return []
-
-    if is_windows:
-        sep = "/"
-    else:
-        sep = os.sep
-
-    if is_windows:
-        path = path.replace("\\", "/")
-
+    sep = os.sep
     parts = path.strip(sep).split(sep)
```

Comment:
It just occurred to me: will the `\\` in `C:\\` lead to issues? e.g.

```
>>> r'C:\\path\to\file'.split('\\')
['C:', '', 'path', 'to', 'file']
```

(there is a `''` item in that list)

Reply:
A raw string expands `\\` into `\\\\` so splitting it over the empty backslash gives an empty string. This is expected. Double backslashes are discouraged in raw strings unless they are very carefully escaped, which IMO is unreadable.

Reply:
Whoops yeah I got confused about something - this should not be an issue.